File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/DnsFactory.java
package tw.nekomimi.nekogram.utils;
import java.net.InetAddress;
import java.util.ArrayList;
import java.util.List;
public class DnsFactory {
    public static InetAddress[] lookup(String domain) throws Exception { return new InetAddress[0]; }
    public static List<String> getTxts(String domain) { return new ArrayList<>(); }
    public static boolean isIpv6Address(String addr) { return addr != null && addr.contains(":"); }
    public static boolean isVPNEnabled() { return false; }
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/DnsFactory.kt
package tw.nekomimi.nekogram.utils

import java.net.InetAddress

object DnsFactory {

    fun providers() = emptyArray<String>()

    @JvmStatic
    @JvmOverloads
    fun lookup(domain: String, fallback: Boolean = false): List<InetAddress> = emptyList()

    @JvmStatic
    fun getTxts(domain: String): List<String> = emptyList()
}
